packages/package_info_plus/package_info_plus_web/lib/package_info_plus_web.dart

```diff
@@ -20,10 +20,30 @@ class PackageInfoPlugin extends PackageInfoPlatform {
     PackageInfoPlatform.instance = PackageInfoPlugin();
   }
 
+  /// Get origin from [uri].
+  ///
+  /// This is different from [Uri.origin] because that does not support
+  /// non- http / https use cases. This function is a workaround,
+  /// specifically for chrome extensions.
+  String origin(Uri uri) {
+    try {
+      return uri.origin;
+    } on StateError catch (e) {
+      String getErrorMessage([bool to = false]) =>
+          'Origin is only applicable ${to ? 'to ' : ''}schemes http and https: chrome-extension://';
+
+      if (e.message.startsWith(getErrorMessage()) ||
+          e.message.startsWith(getErrorMessage(true))) {
```

Comment:
The bool `to` could be difficult to read, as well as the method generating both error messages.

How would you like this?
```suggestion
      const errorPart1 = 'Origin is only applicable';
      const errorPart2 = 'http and https: chrome-extension://';

      if (e.message.contains(errorPart1) && e.message.contains(errorPart2) {
```